### PR TITLE
Clarify strategies for auto-plan

### PR DESCRIPTION
**Pull request by landonreed**
This PR (in pursuit of changes for Call Taker) clarifies/defines the various strategies for replanning a trip whenever a query param change is detected in the form (AKA auto-plan). This includes a config update to autoPlan (no real breaking changes, unless an implementation had `autoPlan: true` at some point, but this was never documented).

The example config shows the approach:
https://github.com/opentripplanner/otp-react-redux/blob/8d500a2a4437843d3ebd6979469b1d7d49de28bd/example-config.yml#L18-L25

### CODE REVIEW

File: lib/actions/form.js [hunk 1]
```diff
@@ -128,3 +130,50 @@ export function formChanged (oldQuery, newQuery) {
     }
   }
 }
+
+/**
+ * Check if the trip should be replanned based on the auto plan strategy,
+ * whether the mobile view is active, and the old/new queries. Response type is
+ * an object containing various booleans.
+ */
+export function checkShouldReplanTrip (autoPlan, isMobile, oldQuery, newQuery) {
```
Comment (evansiroky) [inline lib/actions/form.js:139]: Might be nice to have some unit tests for this. This is non-blocking.

**Review by evansiroky — APPROVED** [review]

File: lib/actions/form.js [hunk 2]
```diff
@@ -128,3 +130,50 @@ export function formChanged (oldQuery, newQuery) {
...
+ */
+const evaluateAutoPlanStrategy = (strategy, fromChanged, toChanged, oneLocationChanged) => {
+  switch (strategy) {
+    case 'ONE_LOCATION_CHANGED':
+      if (oneLocationChanged) return true
+      break
+    case 'BOTH_LOCATIONS_CHANGED':
+      if (fromChanged && toChanged) return true
+      break
+    case 'ANY': return true
```
Comment (binh-dam-ibigroup) [inline lib/actions/form.js:176]: What is this value for? Either remove it or document it in `example-config.yml`.

**Review by binh-dam-ibigroup — APPROVED** [review]
Conditional approval on addressing my comment regarding autoplan strategy `ANY`.

File: lib/actions/form.js (hunk 2 above)
Comment (landonreed) [reply lib/actions/form.js:176]: https://github.com/opentripplanner/otp-react-redux/blob/8d500a2a4437843d3ebd6979469b1d7d49de28bd/example-config.yml#L21-L22

File: lib/actions/form.js (hunk 1 above)
Comment (landonreed) [reply lib/actions/form.js:139]: Added a ticket for revisiting this here: https://github.com/opentripplanner/otp-react-redux/issues/355

**Pull request merged**